realm/realm-library/src/main/java/io/realm/internal/OsCollectionChangeSet.java

```diff
@@ -140,7 +151,9 @@ public boolean isFirstAsyncCallback() {
      * Returns {@code true} if this changeset is empty, and doesn't contain any relevant changes.
      */
     public boolean isEmpty() {
-        return nativeIsEmpty(nativePtr);
+        // Since this wrap a Object Store changeset, it will always contains changes if an
+        // Object Store changeset exists.
+        return nativePtr == 0;
```

Comment:
Is it guaranteed that object store change set cannot be `delete`d without us knowing?

Reply:
Yes, because we create a copy with the changeset here: https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/cpp/observable_collection_wrapper.hpp#L89

And that is only deleted when the finalizer is called.